private-templates-service/client/src/AuthenticatedComponent.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import { Redirect } from "react-router-dom";
+import { UserType } from "./App";
+import Welcome from "./components/Welcome/Welcome";
+
+interface State {
+}
+interface AuthProps {
+  isAuthenticated: boolean,
+  user?: UserType,
+  authButtonMethod: () => Promise<void>
+}
+
+export function requireAuthentication(Component: any) {
+
+  return class AuthenticatedComponent extends React.Component<AuthProps, State> {
+
+    isAuthenticated() {
```

Comment:
I would get in the habit of using ES6 arrow notations for binding.  Also in this case you can simplify it even further:

`isAuthenticated = () => this.props.isAuthenticated`